packages/wdio-cli/src/run.js

```diff
@@ -64,9 +65,28 @@ export default function run (params) {
     })
 }
 
-function launch (wdioConf, params) {
+/**
+ * launcher callback function
+ * @param {number|undefined}    exitCode    exit code
+ * @param {Error|undefined}     err         Error
+ * @param {boolean|undefined}   isComplete  if all tests and hooks were executed
+ */
+export const launchCallback = (exitCode, err, isComplete) => {
```

Comment:
I am not sure if we need this function. Doing `process.exit(code)` was definitely not a great idea. Ideally the `launch` method should end with:

```js
return launcher.run()
```

The reason why processes in Node hang around is that not all event listeners are removed. We should try to ensure that clear up all event listeners so that processes can shutdown gracefully.

Reply:
@christian-bromann we are calling `shutdown` event that ends child processes gracefully, even if there are some listeners it will be killed.
The idea of this callback is to identify if need to call it or no

Reply:
the reason why it was hanging before is that we were not releasing properly STDIN after piping it.